docs/source/deprecation.md

```diff
@@ -0,0 +1,136 @@
+## Deprecation
+
+To help with the deprecation process of entities in leapp, the leapp framework introduced a decorator `deprecated`
+which can be imported from `leapp.utils.deprecation`.
+
+This decorator will mark items as being deprecated and ensures that a deprecation warning is displayed if this item
+is being used.
+
+### Warning message display behaviour
+
+Warning messages are always shown after `snactor` executions via `snactor run` or `snactor workflow run`.
+At the end of executions of `leapp upgrade` or `leapp preupgrade` the warnings are added as report entries with a
+developer audience. If the deprecation is less than 6 months old the severity is set to 'MEDIUM' if it is older than
+6 months the severity is set to 'HIGH'
+
+#### snactor warning message example
+
+```
+============================================================
+                 USE OF DEPRECATED ENTITIES
+============================================================
+
+Usage of deprecated function "deprecated_function" @ /Users/vfeenstr/devel/work/leapp/leapp/tests/data/deprecation-tests/actors/deprecationtests/actor.py:19
+Near:         deprecated_function()
+
+Reason: This function is no longer supported.
+------------------------------------------------------------
+Usage of deprecated Model "DeprecatedModel" @ /Users/vfeenstr/devel/work/leapp/leapp/tests/data/deprecation-tests/actors/deprecationtests/actor.py:20
+Near:         self.produce(DeprecatedModel())
+
+Reason: This model is deprecated - Please do not use it anymore
+------------------------------------------------------------
+Usage of deprecated class "DeprecatedNoInit" @ /Users/vfeenstr/devel/work/leapp/leapp/tests/data/deprecation-tests/actors/deprecationtests/actor.py:21
+Near:         DeprecatedNoInit()
+
+Reason: Deprecated class without __init__
+------------------------------------------------------------
+Usage of deprecated class "DeprecatedBaseNoInit" @ /Users/vfeenstr/devel/work/leapp/leapp/tests/data/deprecation-tests/actors/deprecationtests/actor.py:22
+Near:         DeprecatedNoInitDerived()
+
+Reason: Deprecated base class without __init__
+------------------------------------------------------------
+Usage of deprecated class "DeprecatedWithInit" @ /Users/vfeenstr/devel/work/leapp/leapp/tests/data/deprecation-tests/actors/deprecationtests/actor.py:23
+Near:         DeprecatedWithInit()
+
+Reason: Deprecated class with __init__
+------------------------------------------------------------
+
+============================================================
+             END OF USE OF DEPRECATED ENTITIES
+============================================================
+```
+
+#### leapp report example entries
+
+```
+----------------------------------------
+Risk Factor: medium
+Title: Usage of deprecated class "IsolatedActions" at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/libraries/repofileutils.py:38
+Summary: IsolatedActions are deprecated
+Since: 2020-01-02
```

Comment:
@vinzenz WDYT about print of expected "deadline"? It's known that deprecation process is 6month, so I would put the info that it is expected that functionality will be dropped completely after the period. Something like:
```
Risk Factor: medium
Title: Usage of deprecated class "IsolatedActions" at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/libraries/repofileutils.py:38
Summary: IsolatedActions are deprecated
Since: 2020-01-02
EOL: 2020-07-02
...
```

Reply:
After 6 months it's changing the risk factor to high, that's all I know. I don't know about calling it EOL though.

Reply:
The point is, that after 6months, the decorator and functionality (model/method/...) will be removed the most probably.